### ISSUE TITLE
rate limit error for first allowance

### ISSUE THREAD

**Issue by wp-breeder** [bsm/ratelimit#4]
```go
package main

import (
	"github.com/bsm/ratelimit"

	"fmt"
	"sync"
	"time"
)

func main() {
	wg := &sync.WaitGroup{}
	limiter := ratelimit.New(5, time.Second)
	for i := 0; i < 100; i++ {
		for {
			if !limiter.Limit() {
				break
			}
		}
		wg.Add(1)
		go func(v int) {
			defer func() {
				wg.Done()
			}()
			fmt.Printf("current value:%v time: %v\n", v, time.Now().Unix())

		}(i)
	}
	wg.Wait()
}
```

**Comment by dim** [comment]
@wp-breeder sorry, I cannot see a problem, I have updated your script slightly but it does exactly what it should:

```go
package main

import (
	"github.com/bsm/ratelimit"

	"fmt"
	"sync"
	"time"
)

func main() {
	wg := &sync.WaitGroup{}
	limiter := ratelimit.New(5, time.Second)
	for i := 0; i < 20; i++ {
		for limiter.Limit() {
		}
		wg.Add(1)
		go func(v int) {
			defer wg.Done()
			fmt.Printf("value:%v time:%v\n", v, time.Now().Unix())

		}(i)
	}
	wg.Wait()
}
```

Output: 

```
value:2 time:1626771263
value:1 time:1626771263
value:3 time:1626771263
value:0 time:1626771263
value:4 time:1626771263
value:5 time:1626771264
value:6 time:1626771264
value:7 time:1626771264
value:8 time:1626771264
value:9 time:1626771264
value:10 time:1626771265
value:11 time:1626771265
value:12 time:1626771265
value:13 time:1626771265
value:14 time:1626771265
value:15 time:1626771266
value:16 time:1626771266
value:17 time:1626771266
value:18 time:1626771266
value:19 time:1626771266
```

**Comment by wp-breeder** [comment]
@dim Thanks for replying, but  i running this code , i see that first output  more than five 
go version is  sdk 1.14
os version As shown below：
![image](https://user-images.githubusercontent.com/18710927/126295496-0e60fe63-0fd0-4044-afd3-bb43969761ac.png)

Output:
```text
value:1 time:1626771913
value:3 time:1626771913
value:4 time:1626771913
value:0 time:1626771913
value:2 time:1626771913
value:5 time:1626771913
value:6 time:1626771914
value:7 time:1626771914
value:8 time:1626771914
value:9 time:1626771914
value:10 time:1626771914
value:11 time:1626771915
value:12 time:1626771915
value:13 time:1626771915
value:14 time:1626771915
value:15 time:1626771915
value:16 time:1626771916
value:17 time:1626771916
value:18 time:1626771916
value:19 time:1626771916
```

**Comment by dim** [comment]
That's because it's an overflow rate limit, not a semaphore. Take a look at this:

```
import (
	"fmt"

	"github.com/bsm/ratelimit"

	"time"
)

func main() {
	limiter := ratelimit.New(5, time.Second)
	start := time.Now()
	for i := 0; i < 20; i++ {
		for limiter.Limit() {
		}
	}
	fmt.Println(time.Since(start))
}
```

The output (for me) is `3.000000657s` - which means: 
1. allow 5 to pass through immediately
2. slowly release slots over the next 3s to process the other 15

TLDR: this implementation is very fast, inexpensive and highly accurate "on average", which makes is great for e.g. throttling HTTP requests but not particularly suited for locks with precise time guarantees.

**Comment by wp-breeder** [comment]
@dim , thx
